lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/BuiltinsProcessor.java

```diff
@@ -0,0 +1,211 @@
+package org.enso.interpreter.dsl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openide.util.lookup.ServiceProvider;
+
+import javax.annotation.processing.*;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.*;
+import javax.tools.JavaFileObject;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * The processor used to generate code from the methods of the runtime representations annotated
+ * with {@link Builtin}.
+ *
+ * <p>It represents the first stage in generating the final builtin nodes. The processor will
+ * generate a corresponding @BuiltinMethod class which, in turn, will get processed by another
+ * processor and deliver a RootNode for the method.
+ */
+@SupportedAnnotationTypes("org.enso.interpreter.dsl.Builtin")
+@SupportedSourceVersion(SourceVersion.RELEASE_11)
+@ServiceProvider(service = Processor.class)
+public class BuiltinsProcessor extends AbstractProcessor {
+
+  private static final String BuiltinsPkg = "org.enso.interpreter.node.expression.builtin";
+
+  @Override
+  public final boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+    for (TypeElement annotation : annotations) {
+      Set<? extends Element> annotatedElements = roundEnv.getElementsAnnotatedWith(annotation);
+      for (Element elt : annotatedElements) {
+        if (elt.getKind() == ElementKind.METHOD) {
+          try {
+            handleMethodElement(elt, roundEnv);
+          } catch (IOException ioe) {
+            ioe.printStackTrace();
```

Comment:
Shouldn't we rethrow an error when the generator failed?

Reply:
Possibly report the error via `processingEnv.getMessager().printError(...)` - then it'd be picked up by any tool that invokes the compiler and treated as a compilation error.

Reply:
done